src/RequestHandler.php

```diff
@@ -235,6 +254,26 @@ public function slaveConnected(ConnectionInterface $connection)
         $this->connection->pipe($this->incoming, ['end' => false]);
     }
 
+    /**
+     * Stop the worker if the max execution time has been exceeded and return 504
+     */
+    public function maxExecutionTimeExceeded()
+    {
+        // client went away while waiting for worker
+        if (!$this->connectionOpen) {
+            return false;
+        }
+
+        $response = $this->createErrorResponse('504 Gateway Timeout', 'Maximum execution time exceeded');
+        $this->incoming->write($response);
+        $this->lastOutgoingData = $response;
```

Comment:
Why is this needed? Its not obvious?

Reply:
I mean the assignment to lastoutgoingdata?

Reply:
If its just to not be empty assign ‚not empty‘ or something?

Reply:
If we do not set it then $lastOutgoingData is empty and the if condition in slaveClosed regarding empty data is triggered and it tries to also send the 502 response. Do you know of a better way to handle this?

Reply:
Now that I think of it it actually makes sense to set lastOutgoingData here. Under normal conditions $lastOutgoingData would be set when the worker would send data. Since we are creating a response manually here we also need to set $lastOutgoingData as this is the actual data sent to the client.

In fact we should probably have this for the 503 response too for consistency, though it doesn't really matter because slaveClosed does not trigger there.

Reply:
Something like responseSent or removing the other listeners? Right now its too implicit to be maintainable?

Reply:
Or simply assign ‚not empty‘ to make the purpose clearer.

Reply:
...or pipe the created response into the connection?

Reply:
OK, I changed the assignment to "not empty" and added a comment to make it more clear. I think its more clear now